train.py

```diff
@@ -3,18 +3,19 @@
 import pathlib
 import random
 from dataclasses import dataclass, field
-from typing import Optional
+from typing import Dict, Optional
 
 import torch
 import torch.distributed
-from llama_flash_attn_monkey_patch import replace_llama_attn_with_flash_attn
+import transformers
 from torch.nn import CrossEntropyLoss
+from transformers import LlamaTokenizer, Trainer
 
-replace_llama_attn_with_flash_attn()
+from functionary.prompt import EndToken
+from train.custom_datasets import CustomDataset, split_data
+from train.llama_flash_attn_monkey_patch import replace_llama_attn_with_flash_attn
```

Comment:
Transformers must be imported after the patch. Lets revert back the previous order